### DISCUSSION TITLE
Events for button press, rather than state updates, of a rocker switch

### DISCUSSION THREAD

**Discussion by jamesphw** [Koenkk/zigbee2mqtt#18924] [Devices]
I have an Enbrighten 43080 hard-wired switch installed on my network. I'm using a Sonoff Zigbee "ZBDongle-E" as the contoller, and I'm working in Home Assistant.

The issue I have is that the switch itself exposes an on/off state. On the topic (and in the Z2M logs), there is only something published when the internal state of the switch changes. But I need the button press events, regardless of whether the switch internal state changes.

Strangely this was possible in home assistant ZHA (an event is generated both for the button press and for the state switch), but it does not work on Z2M. How can I get this working on Zigbee2Mqtt?

**Comment by Koenkk** [comment]
See the `state_action` option: https://www.zigbee2mqtt.io/devices/43080.html#options

**Reply by jamesphw** [reply]
Thanks @Koenkk . The state_action option is set to true for the device, but it does not change the behavior (still no event for button presses without a state change).

Is this a bug in the setup for this device? I could perhaps tinker to try and try to fix it, if you could point me to where in the configuration/code I'd need to look.

**Reply by Koenkk** [reply]
I've checked to the code but it should work, so if it isn't it's a bug. The `action` should be added here: https://github.com/Koenkk/zigbee-herdsman-converters/blob/8567f8d7680e04104dc2dc5ee53ae60b3af4cb21/src/converters/fromZigbee.js#L850

**Reply by jamesphw** [reply]
Thanks. If there's information I can collect to help debug, I'm happy to share and then raise this as a bug.

If I have time in the coming weeks I will try to build a debug version of the Z2M + ZHC and see if I can find the issue, but I simply don't know this codebase at all so I may not get far.

**Reply by yanvigdev** [reply]
I also have the same problem with the latest update with TuYa BSD29. I no longer have an on/off button in homeassistant.

**Comment by alexrmay91** [comment]
Just checking to see if you got anywhere with this. I'd really like to capture button press events on this switch, but can't get them either. Also happy to help provide any information I can.

**Comment by jpbaril** [comment]
I opened a ticket as it is still not working for me.
https://github.com/Koenkk/zigbee2mqtt/issues/24400
